src/services/textChanges.ts

```diff
@@ -343,6 +345,13 @@ namespace ts.textChanges {
             this.replaceRange(sourceFile, { pos, end: pos }, createToken(modifier), { suffix: " " });
         }
 
+        public insertTypeAnnotation(sourceFile: SourceFile, node: TypeAnnotatable, type: TypeNode): void {

```

Comment:
so why not just replaceNode with a node with a type annotation?

Reply:
To avoid an issue like #22358 -- we should be adding new text but not reformatting existing text

Reply:
i see.. we should add a comment for that then..